File: core/src/main/kotlin/com/example/toyracers/input/PlayerControlConfig.kt
package com.example.toyracers.input

/** Player-specific input tuning kept separate from vehicle physics. */
data class PlayerControlConfig(
    val steeringSensitivity: Float = 0.85f,
) {
    init {
        require(steeringSensitivity in 0f..1f)
    }

    fun applyTo(input: PlayerInput): PlayerInput = input.copy(steering = input.steering * steeringSensitivity)
}

File: dart/lib/simulation/input/player_control_config.dart
import '../math/float32.dart';
import 'player_input.dart';

/// Player-specific input tuning kept separate from vehicle physics.
final class PlayerControlConfig {
  PlayerControlConfig({double steeringSensitivity = 0.85})
    : steeringSensitivity = Float32.narrow(steeringSensitivity) {
    if (this.steeringSensitivity < 0 || this.steeringSensitivity > 1) {
      throw ArgumentError.value(
        steeringSensitivity,
        'steeringSensitivity',
        'must be in the range [0, 1]',
      );
    }
  }

  final double steeringSensitivity;

  /// Scales only steering; throttle and brake retain their input values.
  PlayerInput applyTo(PlayerInput input) => PlayerInput(
    throttle: input.throttle,
    brake: input.brake,
    steering: Float32.multiply(input.steering, steeringSensitivity),
  );
}
